File: 250620/비오는 날/rainy-day.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        PredictWeatherData[] weatherDataSet = new PredictWeatherData[n];
        for (int i = 0; i < n; i++) {
            String date = sc.next();
            String day = sc.next();
            String weather = sc.next();
            // Please write your code here.

            weatherDataSet[i] = new PredictWeatherData(date, day, weather);
        }

        Arrays.sort(weatherDataSet, (a,b) -> {
            return a.date.compareTo(b.date);
        });

        for (int i = 0; i < n; i++) {
            if (weatherDataSet[i].weather.equals("Rain")) {
                System.out.println(weatherDataSet[i].date + " " + weatherDataSet[i].day + " " + weatherDataSet[i].weather);
                break;
            }
        }
    }

    

    
}

class PredictWeatherData {
    String date;
    String day;
    String weather;

    PredictWeatherData(String date, String day, String weather) {
        this.date = date;
        this.day = day;
        this.weather = weather;
    }
}

// 날짜 순으로 정렬 -> 
// 순차로 돌다가 Rain 찾으면 해당 인덱스의 날짜요일날씨 출력

File: 250923/비오는 날/rainy-day.py
n = int(input())
date = []
day = []
weather = []

ds = []

class D :
    def __init__(self,d,dy,w) :
        self.d = d
        self.dy = dy
        self.w = w

for _ in range(n):
    d, dy, w = input().split()
    date.append(d)
    day.append(dy)
    weather.append(w)

    ds.append(D(d,dy,w))

target_idx = -1

for i,d in enumerate(ds) :
    if d.w == "Rain" :
        if target_idx == -1 :
            target_idx = i
        elif d.d < ds[target_idx].d :
            target_idx = i

print(ds[target_idx].d, ds[target_idx].dy, ds[target_idx].w)



